crates/egui/src/widgets/plot/mod.rs

```diff
@@ -1335,16 +1593,11 @@ impl PreparedPlot {
     fn ui(self, ui: &mut Ui, response: &Response) -> Vec<Cursor> {
         let mut axes_shapes = Vec::new();
 
-        for d in 0..2 {
-            if self.show_axes[d] {
-                self.paint_axis(
-                    ui,
-                    d,
-                    self.show_axes[1 - d],
-                    &mut axes_shapes,
-                    self.sharp_grid_lines,
-                );
-            }
+        if self.show_grid.x {
+            self.paint_grid::<{ X_AXIS }>(ui, &mut axes_shapes);
```

Comment:
What weird syntax is this? O.o

Will this work: ?
```suggestion
            self.paint_grid::<X_AXIS>(ui, &mut axes_shapes);
```